src/matlab2tikz.m

```diff
@@ -950,7 +956,7 @@ function addCustomCode(fid, before, code, after)
             % Make sure that m2t.legendHandles is a row vector.
             for lhandle = m2t.legendHandles(:)'
                 ud = get(lhandle, 'UserData');
-                if isVisible(lhandle) && any(handle == ud.handle)
+                if isAxisVisible(lhandle) && any(handle == ud.handle)
```

Comment:
Probably we should rename `isAxisVisible` to `isVisibleContainer` to convey the meaning better.